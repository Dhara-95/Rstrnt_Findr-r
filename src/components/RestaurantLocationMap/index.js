import React from "react";
import { MapContainer, TileLayer, Marker, Popup } from "react-leaflet";

const RestaurantLocationMap = (props) => {
  const { latitude, longitude } = props.coordinates;
  const { name } = props;

  return (
    <MapContainer
      className="mx-auto"
      center={[latitude, longitude]}
      zoom={13}
      scrollWheelZoom={false}
    >
      <TileLayer
        attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
      />
      <Marker position={[latitude, longitude]}>
        <Popup>{name}</Popup>
      </Marker>
    </MapContainer>
  );
};

export { RestaurantLocationMap };
