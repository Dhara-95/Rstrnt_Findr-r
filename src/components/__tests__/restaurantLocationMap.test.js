import React from "react";
import renderer from "react-test-renderer";

import { RestaurantLocationMap } from "../RestaurantLocationMap";

it("displays a map with restaurant location", () => {
  const tree = renderer
    .create(
      <RestaurantLocationMap
        coordinates={{ latitude: 51.7375787, longitude: -0.4688423 }}
        name={"Apsley Chinese"}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
