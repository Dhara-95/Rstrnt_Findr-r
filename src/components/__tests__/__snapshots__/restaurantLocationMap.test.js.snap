// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a map with restaurant location 1`] = `
<div
  className="mx-auto"
/>
`;
